zerver/webhooks/taiga/view.py

```diff
@@ -278,25 +298,28 @@ def generate_content(data: Mapping[str, Any]) -> str:
     """ Gets the template string and formats it with parsed data. """
     template = templates[data['type']][data['event']]
     content = template.format(**data['values'])
-    end_type = 'end_type'
-    if template.endswith('{subject}**.'):
-        end_type = 'subject'
-    elif template.endswith('{epic_subject}**.'):
-        end_type = 'epic_subject'
-    elif template.endswith('{new}.') or template.endswith('{new}**.'):
-        end_type = 'new'
-    elif template.endswith('{old}.') or template.endswith('{old}**.'):
-        end_type = 'old'
-    end = data['values'].get(end_type)
-
-    if end[-1] in string.punctuation:
-        content = content[:-1]
-
     return content
 
 def get_owner_name(message: Mapping[str, Any]) -> str:
     return message["by"]["full_name"]
 
+def get_owner_link(message: Mapping[str, Any]) -> str:
+    return message['by']['permalink']
+
 def get_subject(message: Mapping[str, Any]) -> str:
     data = message["data"]
-    return data.get("subject", data.get("name"))
+    if 'permalink' in data:
+        return '[' + data.get('subject', data.get('name')) + ']' + '(' + data['permalink'] + ')'
```

Comment:
@timabbott Sorry I didn't mention it in the PR clearly but the PR actually adds links to issues/ epics etc. I have added a few tests for them and also the role of the functions at the end of views.py file is to use issue/sprint name with link if possible else just use the name. The code which I have highlighted is one of the functions.